daemon/src/file_asset_source.rs

```diff
@@ -394,233 +446,278 @@ impl FileAssetSource {
-        {
-            let txn = self.db.ro_txn()?;
-            for result in self.iter_metadata(&txn)? {
-                let (path, metadata) = result?;
-                let metadata = metadata.get()?;
-                let changed = if let Some(importer) = self.importers.get_by_path(&path) {
-                    metadata.get_importer_version() != importer.version()
-                        || metadata.get_importer_options_type()?
-                            != importer.default_options().uuid()
-                        || metadata.get_importer_state_type()? != importer.default_state().uuid()
-                        || metadata.get_importer_type()? != importer.uuid()
-                } else {
-                    false
-                };
+
+    fn check_for_importer_changes(&self) -> bool {
+        let txn = self.db.ro_txn().expect("Failed to open ro txn");
+
+        let changed_paths: Vec<PathBuf> = self
+            .iter_metadata(&txn)
+            .filter_map(|(path, metadata)| {
+                let metadata = metadata.get().ok()?;
+                let changed = self
+                    .importers
+                    .get_by_path(&path)
+                    .and_then(|importer| {
+                        let importer_version = metadata.get_importer_version();
+                        let options_type = metadata.get_importer_options_type().ok()?;
+                        let state_type = metadata.get_importer_state_type().ok()?;
+                        let importer_type = metadata.get_importer_type().ok()?;
+
+                        Some(
+                            importer_version != importer.version()
+                                || options_type != importer.default_options().uuid()
+                                || state_type != importer.default_state().uuid()
+                                || importer_type != importer.uuid(),
+                        )
+                    })
+                    .unwrap_or(false);
+
                 if changed {
-                    changed_paths.push(path);
+                    Some(path)
+                } else {
+                    None
                 }
-            }
-        }
-        if !changed_paths.is_empty() {
-            let mut txn = self.db.rw_txn()?;
-            for path in changed_paths {
-                self.tracker.add_dirty_file(&mut txn, &path)?;
-            }
-            txn.commit()?;
-            return Ok(true);
+            })
+            .collect();
+
+        let has_changed_paths = !changed_paths.is_empty();
+        if has_changed_paths {
+            // TODO(happens): Log errors here
+            let mut txn = self.db.rw_txn().expect("Failed to open rw txn");
+            changed_paths.iter().for_each(|p| {
+                // TODO(happens): Handle this error
+                self.tracker
+                    .add_dirty_file(&mut txn, &p)
+                    .unwrap_or_else(|err| error!("Failed to add dirty file, {}", err));
```

Comment:
Are these TODOs outdated? seems like it logs fine. The logging is fine - add_dirty_file seems to read from FS which can have transient errors which we can just log and continue execution.